python-package/lightgbm/basic.py

```diff
@@ -2823,14 +2823,14 @@ def predict(self, data, start_iteration=None, num_iteration=None,
         data : string, numpy array, pandas DataFrame, H2O DataTable's Frame or scipy.sparse
             Data source for prediction.
             If string, it represents the path to txt file.
-        start_iteration : int or None, optional (default=None)
+        start_iteration : int, optional (default=0)
```

Comment:
In `num_iteration` None is useful because it allows to set `best_iteration` implicitly. But for `start_iteration` I don't see any reasons to allow `None`. Actually, it was just an alias for `<=0`. I believe it doesn't worth to maintain `None`.